client/src/main/scripts/utils/databaseAPI.ts

```diff
@@ -142,17 +142,17 @@ function fetchFact(
     });
 }
 
-export interface VideoFactHashMap {
-  hash: string;
-  videoFact: Foundation.VideoFactContent;
-}
+export type FactHashMap = [
+  string,
+  Foundation.VideoFactContent | Foundation.DocumentFactContent
+];
 
```

Comment:
This whole file seems like it should be dead, and replaced by `FoundationData`.

Reply:
I agree. But several other components touch it, so it would be a larger refactor.

Reply:
Considering:

```
class FoundationDataBuilder {
  requestedFacts: Set<string> = new Set();

  // es6 syntax
  add = (hash: string): void => {
    this.requestedFacts.add(hash);
  };
 
  // es5 syntax
  addEs5(hash: string): void {
    this.requestedFacts.add(hash);
  }
```
}

The technical differences are:

`add` can never be unbound from `this` as the instance of the class.
`addEs5` is added to the prototype. So `addEs5` doesn’t show up in `Object.getOwnPropertyNames(new FoundationDataBuilder())`.  But it does in `Object.getOwnPropertyNames(new FoundationDataBuilder().__proto__)`

Reply:
I didn't know what this means, so I did some digging.  It looks like `method = ()` is slower than `method()` by a significant margin (~7x slower).  WIth `method()`, there is just one function which is shared by every class.  With `method = ()`, every single instance of the class gets its own method, which takes more memory and is more work for the JIT, because it never realizes that the method is really the same. ([search "performance"](https://medium.com/@charpeni/arrow-functions-in-class-properties-might-not-be-as-great-as-we-think-3b3551c440b1)).

The one exception is a react class component, for the case where a method is passed as a prop.  If a react class component has a method which will be passed as a prop, [JSX breaks the "this"](https://www.freecodecamp.org/news/this-is-why-we-need-to-bind-event-handlers-in-class-components-in-react-f7ea1a6f93eb/#:~:text=When%20we%20bind%20the%20this,scope%20they%20are%20defined%20in.), which is fixed by `method = ()`.

There aren't many `FoundationData` instances, so the performance hit of `method = ()` isn't a big deal.  But for transcript parsing and manipulation, it likely would be a big deal.  I think our codestyle should be `method = ()` for React classes where we are passing the method as a prop, but `method() {` everywhere else.